### DISCUSSION TITLE
How to enable letsencrypt certificate per app's domain?

### DISCUSSION THREAD

**Discussion by beydogan** [dokku/dokku#6230] [Q&A]
I'm looking for a way to create and issue individual certificate per app's domain. 

My use case is; I have a SAAS like a website builder where customers point their domains to my server and I serve the content. But they might not really do it all the time. When they set their domain, I do `dokku domains:add`, then I would do `letsencrypt:enable` but if their DNS is not updated or valid it will fail. Also the SSL certificate will have list of all domains which I want to avoid.

It is kind of possible achieve this by editing the letsencrypt plugin code, could you at least point me a direction about it where to start? 

I also need to create nginx config per domain to set the SSL related fields, is it possible to do this using `nginx.conf.sigil` ?

**Comment by josegonzalez** [comment]
You can switch to the openresty proxy plugin and let that do automatic letsencrypt integration on first request. Works fairly similar the existing nginx plugin.
